src/commands/moderation/clear.ts

```diff
@@ -0,0 +1,78 @@
+import { ApplicationCommandOptionType, ApplicationCommandType, TextChannel, Colors } from 'discord.js';
+import { slashCommand, slashCommandStructure, makeEmbed, constantsConfig } from '../../lib';
+
+const data = slashCommandStructure({
+    name: 'clear',
+    description: 'Delete a specified number of messages in the current channel.',
+    type: ApplicationCommandType.ChatInput,
+    default_member_permissions: constantsConfig.commandPermission.MANAGE_SERVER,
+    dm_permission: false,
+    options: [{
+        name: 'amount',
+        description: 'Number of messages to delete (1-100).',
+        type: ApplicationCommandOptionType.Integer,
+        required: true,
+        min_value: 1,
+        max_value: 100,
+    }],
+});
+
+export default slashCommand(data, async ({ interaction, log, client }) => {
+    const amount = interaction.options.getInteger('amount');
+
+    if (!interaction.channel || !interaction.channel.isTextBased()) {
+        return interaction.reply({ content: 'This command can only be used in text channels.', ephemeral: true });
+    }
+
+    if (amount === null) {
+        return interaction.reply({ content: 'Please specify the number of messages to delete.', ephemeral: true });
+    }
+
+    try {
+        const messages = await interaction.channel.bulkDelete(amount, true);
+
+        const replyEmbed = makeEmbed({
+            title: 'Messages Deleted',
+            description: `Successfully deleted **${messages.size}** messages.`,
+            color: Colors.Green,
+            footer: { text: `Requested by ${interaction.user.tag}`, iconURL: interaction.user.displayAvatarURL() },
+            timestamp: new Date(),
+        });
+
+        const targetChannel = await client.channels.fetch(constantsConfig.channels.MOD_LOGS);
```

Comment:
We usually retrieve the `#mod-logs` channel through the [`GuildChannelManager.resolve()`](https://discord.js.org/docs/packages/discord.js/14.15.2/GuildChannelManager:Class#resolve) method.
For consistency, I would do the same here:
```suggestion
        const targetChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
```

Reply:
I would also suggest changing `targetChannel` to `modLogsChannel` for consistency too. It also makes it clear what this const refers to.